### ISSUE TITLE
T5 Pretraining and Finetuning for Custom Language

### ISSUE THREAD

**Issue by ashispapu** [google-research/text-to-text-transfer-transformer#372]
Hi,

Thanks for sharing the T5 pre training examples. I have followed the same for another language. I have two tasks in the pre training phase(unsupervised and question generation tasks). Then I tried to fine tune the pre trained model for question generation only. However it is not able produce good results on unseen/test data. I have checked the training loss which has reduced significantly and validated on the train data and it does learn that.

Do you have any suggestions to improve it on fine tuning task. And is there any better way/criteria to validate the pre trained model before starting the fine tuning of tasks. I could see pre trained model loss is in the range of 0.15 to 0.50 .

**Comment by craffel** [comment]
Hi, there's really many reasons that things could be going wrong and it's a bit hard to investigate for you. Is your pre-training dataset so small that you are overfitting to it? Is the pre-training data noisy? Is the QA dataset for fine-tuning small? Noisy? Is the model producing answers that are wrong, or is outputting gibberish? These are all worth looking in to, but unfortunately we don't have the bandwidth to help debug. Good luck!

**Comment by ashispapu** [comment]
Thanks @craffel  for you inputs. Could you also share what would be a good amount of sentences for pretraining for a new language, if you have any such observations .

**Comment by craffel** [comment]
See section 3.4.2 of https://arxiv.org/pdf/1910.10683.pdf tl;dr, we found that if you repeat your pre-training data more than ~100x over the course of pre-training it can hurt you downstream. So if your pre-train the model on batches containing a total of 100B tokens, you should have a pre-training dataset with at least 1B tokens in it (as a rough rule of thumb).

**Comment by ashispapu** [comment]
Thanks @craffel for the response. I guess the 1B tokens you mentioned is for the default pre-training(T5-base). Would that also hold good for T5(small and large), given the different scale of complexities of the network architectures.

**Comment by craffel** [comment]
All model sizes were trained for 1T (not B) tokens.
